### ISSUE TITLE
ALC256 on macOS Sierra on Kabylake

### ISSUE THREAD

**Issue by MuntashirAkon** [insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12#19]
This is a bit similar to the other [issue here](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/issues/11), but since the Subsystem Id is different, I am opening a separate issue for this.

I have installed the [`AppleHDA-276.26-ALC256_v2_-_Working_Combojack`](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/blob/master/Laptop/ALC256_v2%20-%20Working%20Combojack.zip?raw=true) found in the repo, which worked but only partially. It recognizes my headphones but no sound. Other things just works fine.

1. [codec_dump#0.txt](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/files/1048730/codec_dump.0.txt)
2. [Muntashir’s MacBook Air.ioreg.zip](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/files/1048734/Muntashir.s.MacBook.Air.ioreg.zip)
3. [screen shot 2017-06-03 at 1 25 07 am.jpg](https://cloud.githubusercontent.com/assets/8176128/26741528/87cc2b7c-47fb-11e7-9fe7-a3fd84e27eeb.png)

I'm running macOS Sierra 10.12.1 on Dell Inspiron 15-5567 (i3-7100u with HD620).

**Comment by K0guma** [comment]
Read #11 please

**Comment by K0guma** [comment]
Plus, you should upgrade.

**Comment by MuntashirAkon** [comment]
@K0guma Can you please briefly explain what solved the problem. I've tried all the versions of AppleHDA there, not a single one worked properly. I've also tried installing the `FakPCIID_Intel_HDMI_Audio.kext` (where there's already my device id (`8086:9d71`) is present) along with mirone's `aDummyHDA.kext` with `layout-id: 3`: it didn't work.

**Comment by K0guma** [comment]
use the newest fakepciid and fakepciid_Intel_HDMI_audio.kext from rehabman bitbucket. Then use the applehda from the post.

**Comment by K0guma** [comment]
the V2

**Comment by MuntashirAkon** [comment]
Solved using [AppleHDA-279.48-ALC256.zip](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/files/1049678/AppleHDA-279.48-ALC256.zip). Thanks @K0guma for your help.

**Comment by MuntashirAkon** [comment]
Headphones are still not working after a reboot.

**Comment by MuntashirAkon** [comment]
Ha, ha, this is getting a little funnier. Headphones are working after another reboot :1st_place_medal:

**Comment by MuntashirAkon** [comment]
For future reference, this is the kext that worked for me: [AppleHDA-279.48-ALC256.zip](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/files/1049741/AppleHDA-279.48-ALC256.zip)

**Comment by MuntashirAkon** [comment]
Does `Internal Microphone` means `Internal Audio`? Microphones doesn't work. But when I select `Headphones`, it records internal audio (How amazing is this!).

**Comment by insanelydeepak** [comment]
hi you need to use Codec_commander in order to get working External Mic

**Comment by insanelydeepak** [comment]
AppleHDA-276.26-ALC256_v2_-_Working_Combojack is for you as per codec_dump , in order to fix headphone you need to use CodecCommander , if still dont works for you , then make custom config for CC

**Comment by MuntashirAkon** [comment]
Ok, let me try again. BTW, I'm interested in using dummy kext instead of directly patching AppleHDA, can you help me with that?

**Comment by insanelydeepak** [comment]
> I'm interested in using dummy kext instead of directly patching AppleHDA, can you help me with that?

CloverHDA is what you are looking for

**Comment by MuntashirAkon** [comment]
I know, but I don't have the required files to do so (that you've used for `AppleHDA-276.26-ALC256_v2_-_Working_Combojack`). It would be very good if you helped me.

**Comment by insanelydeepak** [comment]
> I don't have the required files to do so 
which files

**Comment by MuntashirAkon** [comment]
Layout13.xml.zlib and Platforms.xml.zlib, maybe, or can I just use the one that is available in the `AppleHDA-276.26-ALC256_v2_-_Working_Combojack`?

**Comment by MuntashirAkon** [comment]
Oh, I see, you've updated the repo. Thanks.

**Comment by MuntashirAkon** [comment]
Weird! Headphones aren't working with this patch, but microphone's working. And with the patched AppleHDA that I've posted [earlier](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/issues/19#issuecomment-305988431), headphone's working but microphone's don't! (although I think this kext isn't for a combo jack since I've had to switch them manually). BTW, I'm still testing, let's see if there's any miracle to happen after some numbers of reboot :)

**Comment by MuntashirAkon** [comment]
Still doesn't work after several reboots, rebuilding kernel cache, etc.
[Muntashir’s MacBook Air.ioreg.zip](https://github.com/insanelydeepak/cloverHDA-for-Mac-OS-Sierra-10.12/files/1051895/Muntashir.s.MacBook.Air.ioreg.zip)

![screen shot 2017-06-05 at 5 13 35 pm](https://cloud.githubusercontent.com/assets/8176128/26782147/e7b621b8-4a13-11e7-8221-e4b6351f10bb.png)

**Comment by insanelydeepak** [comment]
Try codec_commander

**Comment by MuntashirAkon** [comment]
Oh shit! This was a new install, I forgot to install the `CodecCommander`! Everything's now working perfectly. Thanks for everything.

BTW, you should include in the `readme.txt` about this issue (ie. headphone sound can only work upon the installation of CodecCommander).

**Comment by insanelydeepak** [comment]
>  headphone sound can only work upon the installation of CodecCommander

done 

feel free to make pull request , if you want to update something
